Fix warnings

NathanBnm opened the pull request:
Hi! I figured out that there were some warnings when building the app. Some things were deprecated so I fixed them.

Here were the warning messages:

```bash
../src/Wallpaperize.vala:27.31-27.56: warning: Gdk.Screen.get_primary_monitor has been deprecated since 3.22
../src/Wallpaperize.vala:29.9-29.35: warning: Gdk.Screen.get_monitor_geometry has been deprecated since 3.22
../src/Wallpaperize.vala:31.29-31.59: warning: Gdk.Screen.get_monitor_scale_factor has been deprecated since 3.22
../src/Application.vala:31.9-31.17: warning: Granite.Application.app_years has been deprecated since 0.4.2. Use 
../src/Application.vala:36.9-36.16: warning: Granite.Application.app_icon has been deprecated since 0.4.2. Use 
../src/Application.vala:37.9-37.16: warning: Granite.Application.main_url has been deprecated since 0.4.2. Use 
../src/Application.vala:38.9-38.15: warning: Granite.Application.bug_url has been deprecated since 0.4.2. Use 
../src/Application.vala:39.9-39.16: warning: Granite.Application.help_url has been deprecated since 0.4.2. Use 
../src/Application.vala:40.9-40.21: warning: Granite.Application.translate_url has been deprecated since 0.4.2. Use 
../src/Application.vala:41.9-41.21: warning: Granite.Application.about_authors has been deprecated since 0.4.2. Use 
../src/Application.vala:42.9-42.25: warning: Granite.Application.about_translators has been deprecated since 0.4.2. Use 
../src/Application.vala:44.9-44.26: warning: Granite.Application.about_license_type has been deprecated since 0.4.2. Use 
Compilation succeeded - 12 warning(s)
```

Please only merge this PR after merging my last one about translations otherwise you may have some conflicts.

**This PR is ready for review**

NathanBnm committed:
Make the app translatable

NathanBnm committed:
French translations

NathanBnm committed:
Improve "drop down menu" to "context menu"

NathanBnm committed:
Fix Gtk.Screen warnings

NathanBnm committed:
Fix Gtk.Screen warnings

NathanBnm committed:
Fix error

NathanBnm committed:
Fix error

NathanBnm committed:
Remove old method

NathanBnm committed:
Remove old method

NathanBnm committed:
Remove deprecated attributes

NathanBnm committed:
Remove deprecated attributes

NathanBnm commented:
EDIT : I didn't expect that my previous commits from my other branch will be there too. As you want you can merge this to apply both this PR and #18 and close #18. Or you can still merge #18 and then merge this PR too. Sorry for duplicate.

ryonakano commented:
@NathanBnm It seems you checked out your `fix-warnings` branch from your `add-translations` branch, not from `master` branch. This makes this PR looks like as if it has many changed files. I think you should change the base branch in order to keep this PR simple with the following command:

```
git rebase --onto master add-translations fix-warnings
git push origin fix-warnings
```

ryonakano commented:
Also, because the class `Granite.Application` is deprecated (see https://github.com/elementary/granite/commit/2b898176031872a7547b1f51cfcb8659f8ecdb45), we may want to change it to Gtk.Application. Also, because Gtk.Application does not have fields like `program_name` or `app_launcher`, you can remove them. See the diff below to see what will change:

```diff
-public class Wallpaperize.Application : Granite.Application {
-    public const string PROGRAM_ID = "com.github.philip-scott.wallpaperize";
+public class Wallpaperize.Application : Gtk.Application {
     public const string PROGRAM_NAME = "Wallpaperize";
 
     construct {
         flags |= ApplicationFlags.HANDLES_OPEN;
-
-        application_id = PROGRAM_ID;
-        program_name = PROGRAM_NAME;
-        exec_name = PROGRAM_ID;
-        app_launcher = PROGRAM_ID;
-        build_version = "1.0";
+        application_id = "com.github.philip-scott.wallpaperize";
     }
```

(I also removed the constant `PROGRAM_ID` because it is never used more than twice once we remove those fields)

NathanBnm commented:
> @NathanBnm It seems you checked out your `fix-warnings` branch from your `add-translations` branch, not from `master` branch. This makes this PR looks like as if it has many changed files. I think you should change the base branch in order to keep this PR simple with the following command:
> 
> ```
> git rebase --onto master add-translations fix-warnings
> git push origin fix-warnings
> ```

Oh yes you're right, I didn't noticed that, thanks for reporting. I'm going to check that !

NathanBnm committed:
Merge branch 'fix-warnings' of https://github.com/NathanBnm/wallpaperize into fix-warnings
Fixes

NathanBnm committed:
Revert translations for this branch

NathanBnm committed:
Fix

NathanBnm committed:
Applied diff suggested by @ryonakano

NathanBnm commented:
@ryonakano I was unable to rebase branch, I don't really now why. But I just reverted changes from the other branch and this is better now. I added your diff too :smile: 

I'm still learning, mistakes can happen :-1:

ryonakano commented on src/Application.vala:
```diff
@@ -19,29 +19,13 @@
 
     construct {
         flags |= ApplicationFlags.HANDLES_OPEN;
 
```
You may want to remove this extra new line, because there are only two lines in this constructor now.

NathanBnm replied:
I don't really understand I already made these changes

ryonakano replied:
I'm sorry for my lack of explanation.

After you addressed my changes suggestion, line 25 to 29 of `Application.vala` should look like this:

```vala
    construct {
        flags |= ApplicationFlags.HANDLES_OPEN;

        application_id = "com.github.philip-scott.wallpaperize";
    }
```

Although this works well, I think it would be even better to remove the blank line (line 27) like this:

```vala
    construct {
        flags |= ApplicationFlags.HANDLES_OPEN;
        application_id = "com.github.philip-scott.wallpaperize";
    }
```

NathanBnm replied:
Yes sure I'll fix this. I didn't understand you were speaking about blank lines :smile:

NathanBnm replied:
This should be better now

ryonakano replied:
Yes, it looks better. Thank you!

NathanBnm committed:
Remove blank

ryonakano commented on src/Application.vala:
```diff
@@ -19,29 +19,12 @@
-public class Wallpaperize.Application : Granite.Application {
-    public const string PROGRAM_ID = "com.github.philip-scott.wallpaperize";
+public class Wallpaperize.Application : Gtk.Application {
     public const string PROGRAM_NAME = "Wallpaperize";
```
BTW I guess we can also remove this line since the property `program_name` that uses this constant get removed in this PR and as far as I [searched](https://github.com/Philip-Scott/wallpaperize/search?q=PROGRAM_NAME&unscoped_q=PROGRAM_NAME) it is not used in elsewhere.

NathanBnm replied:
@ryonakano this constant is used in the main function as following:

```vala
Environment.set_application_name (PROGRAM_NAME);
Environment.set_prgname (PROGRAM_NAME);
```

ryonakano replied:
Ah my bad, I should have checked the whole code deeper…😅

NathanBnm commented:
@Philip-Scott any news?

NathanBnm committed:
Merge https://github.com/Philip-Scott/wallpaperize into fix-warnings

NathanBnm commented:
@Philip-Scott would tout please mind to review and merge this?

Philip-Scott reviewed (approved):
The code looks good, I'd like to test it before merging but It should work correctly. Sorry, I turned off all Github notifications for this project a while back, and honestly just getting that motivation/energy to do coding related side-projects been hard lately.

NathanBnm commented:
No worries it would be awesome if you could take the time to review the PRs for this project 😁
